blockly-ws-search/src/WorkspaceSearch.js

```diff
@@ -0,0 +1,660 @@
+/**
+ * @license
+ * Copyright 2020 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @fileoverview Object responsible for workspace search.
+ * @author aschmiedt@google.com (Abby Schmiedt)
+ * @author kozbial@google.com (Monica Kozbial)
+ */
+
+
+import { injectSearchCss } from './css.js';
+import * as Blockly from 'blockly/core';
+
+export class WorkspaceSearch {
+  /**
+   * Class for workspace search.
+   * @param {!Blockly.WorkspaceSvg} workspace
+   */
+  constructor(workspace) {
+    /**
+     * The workspace the search bar sits in.
+     * @type {!Blockly.WorkspaceSvg}
+     * @private
+     */
+    this.workspace_ = workspace;
+
+    /**
+     * HTML container for the search bar.
+     * @type {?HTMLElement}
+     * @private
+     */
+    this.htmlDiv_ = null;
+
+    /**
+     * The div that holds the search bar actions.
+     * @type {?HTMLElement}
+     * @protected
+     */
+    this.actionDiv_ = null;
+
+    /**
+     * The text input for the search bar.
+     * @type {?HTMLInputElement}
+     * @private
+     */
+    this.inputElement_ = null;
+
+    /**
+     * The placeholder text for the search bar input.
+     * @type {string}
+     * @private
+     */
+    this.textInputPlaceholder_ = 'Search';
```

Comment:
Just a thought, and not merge blocking, but what's the localization story here? Can this be integrated with Blockly.Msg.

Reply:
Since that is the only translation that would need to be done, we side stepped the issue a bit by having a setSearchPlaceholder so developers can set it. We can look into this more if people are interested, however we were not originally planning on it.